bdb/phys_rep_lsn.c

```diff
@@ -0,0 +1,79 @@
+#include <build/db.h>
+#include <strings.h>
+
+#include "bdb_int.h"
+#include "phys_rep_lsn.h"
+#include "locks.h"
+
+extern bdb_state_type *bdb_state;
+
+LOG_INFO get_last_lsn(bdb_state_type* bdb_state)
+{
+    int rc; 
+
+    /* get db internals */
+    DB_LOGC* logc;
+    DBT logrec;
+    DB_LSN last_log_lsn;
+    LOG_INFO log_info;
+    log_info.file = log_info.offset = log_info.size = 0;
+
+    rc = bdb_state->dbenv->log_cursor(bdb_state->dbenv, &logc, 0);
+    if (rc) {
+        logmsg(LOGMSG_ERROR, "%s: can't get log cursor rc %d\n", __func__, rc);
+        return log_info;
```

Comment:
no- log_info was declared on the stack- you can't return it.  Refactor this function to take a pointer to the returned lsn.  So, int get_last_lsn(bdb_state_type *bdb_state, LOG_INFO *loginfo)